slack_lambda/main.tf

```
@@ -19,8 +19,13 @@ locals {
         eventmsg = event['Records'][0]['Sns']['Message']
         try:
           data = json.loads(eventmsg)
-          if data['detailType'] == 'CodePipeline Pipeline Execution State Change':
+          if 'detailType' in data and data['detailType'] == 'CodePipeline Pipeline Execution State Change':
             msgtext = 'auto-terraform:  ' + data['detail']['pipeline'] + ' pipeline ' + data['detail']['state'] + ' with execution ID ' + data['detail']['execution-id']
+          elif 'AlarmName' in data and 'AlarmDescription' in data:
+            msgtext = ('Alarm has gone off!\n' + data['AlarmName'] + '\n' + data['AlarmDescription'] + '\n' + data['NewStateReason'] + '\nTime: ' +
+                      data['StateChangeTime'] + '\nRegion: ' + data['Region'])
```

idk if this would be easier to read/follow?

```suggestion
            msgtext = '\n'.join(' '.join(pair) for pair in [
              ['Alarm has gone off!'],
              [data['AlarmName'], data['AlarmDescription']],
              [data['NewStateReason']],
              ['Time:', data['StateChangeTime']],
              ['Region:', data['Region']]))
```

---

bonus...add some `*` for bold text? 💅  checked via https://api.slack.com/docs/messages/builder

```suggestion
```suggestion
            msgtext = '\n'.join(' '.join(pair) for pair in [
              ['*Alarm has gone off!*'],
              ['*' + data['AlarmName'] + '*', data['AlarmDescription']],
              [data['NewStateReason']],
              ['*Time*:', data['StateChangeTime']],
              ['*Region*:', data['Region']])
``````

---

Thanks!